examples/standalone/runfile/dynamics.py

```diff
@@ -1,120 +1,152 @@
-    fv3core.set_rebuild(False)
-
-    # namelist setup
-    spec.set_namelist(data_dir + "/input.nml")
-
-    nml2 = yaml.safe_load(
-        open(
-            data_dir + "/input.yml",
-            "r",
+    timer = Timer()
+    timer.start("total")
+    with timer.clock("init"):
+
+        usage = "usage: python %(prog)s <data_dir> <timesteps> <backend>"
+        parser = ArgumentParser(usage=usage)
+
+        parser.add_argument(
+            "data_dir",
+            type=str,
+            action="store",
+            help="directory containing data to run with",
         )
-    )["namelist"]
-    experiment_name = yaml.safe_load(
-        open(
-            data_dir + "/input.yml",
-            "r",
+        parser.add_argument(
+            "time_step",
+            type=int,
+            action="store",
+            help="number of timesteps to execute",
         )
-    )["experiment_name"]
-
-    # set up of helper structures
-    serializer = serialbox.Serializer(
-        serialbox.OpenModeKind.Read,
-        data_dir,
-        "Generator_rank" + str(rank),
-    )
-    cube_comm = util.CubedSphereCommunicator(
-        comm, util.CubedSpherePartitioner.from_namelist(nml2)
-    )
-
-    # get grid from serialized data
-    grid_savepoint = serializer.get_savepoint("Grid-Info")[0]
-    grid_data = {}
-    grid_fields = serializer.fields_at_savepoint(grid_savepoint)
-    for field in grid_fields:
-        grid_data[field] = serializer.read(field, grid_savepoint)
-        if len(grid_data[field].flatten()) == 1:
-            grid_data[field] = grid_data[field][0]
-    grid = fv3core.testing.TranslateGrid(grid_data, rank).python_grid()
-    spec.set_grid(grid)
-
-    # set up grid-dependent helper structures
-    layout = spec.namelist.layout
-    partitioner = util.CubedSpherePartitioner(util.TilePartitioner(layout))
-    communicator = util.CubedSphereCommunicator(comm, partitioner)
-
-    # create a state from serialized data
-    savepoint_in = serializer.get_savepoint("FVDynamics-In")[0]
-    driver_object = fv3core.testing.TranslateFVDynamics([grid])
-    input_data = driver_object.collect_input_data(serializer, savepoint_in)
-    input_data["comm"] = communicator
-    state = driver_object.state_from_inputs(input_data)
-
-    # warm-up timestep
-    fv_dynamics.fv_dynamics(
-        state,
-        communicator,
-        input_data["consv_te"],
-        input_data["do_adiabatic_init"],
-        input_data["bdt"],
-        input_data["ptop"],
-        input_data["n_split"],
-        input_data["ks"],
-    )
-    timing.time("init")
-
-    # Run the dynamics
-    timing.time("main_loop")
-    for i in range(time_step - 1):
+        parser.add_argument(
+            "backend",
+            type=str,
+            action="store",
+            help="path to the namelist",
+        )
+        args = parser.parse_args()
+        backend = args.backend
+        data_dir = args.data_dir
+        time_step = args.time_step
```

Comment:
Is it necessary to unpack these? If you use e.g. `args.backend` below, then it's clearer when reading e.g. line 97 that "backend" comes from the command-line arguments.